docs/quickstart.md

```diff
@@ -0,0 +1,173 @@
+Click the small _Click here_ link at the bottom of the page to proceed and create a repository named `repo` (we don't want to add the sample data for this guide):
+
+![](_images/quickstart-lakefs-sample-repo.png)
+
+??? tip "Tip: Creating a repository later"
+
+    If you have inadvertently skipped over the quickstart repository creation page, you can always create a new repository on the [_Repositories_ tab](http://localhost:8000/repositories) in the lakeFS web UI (and optionally choose to add the sample data):
+
+    ![](_images/quickstart-lakefs-repositories.png)
+
+!!! success
+    You have successfully created a lakeFS repository named `repo`, ready to be used with `lakefs-spec`.
+
+### Using the lakeFS `fsspec` file system
+
+While the `quickstart` repository already contains some files, we won't be using them for the remainder of this guide. Instead, we will now use the `lakefs-spec` file system interface to upload a file to the repository we just created, make a commit, and read back the committed data.
+
+To get started, create a file called `quickstart.py` with the following contents:
+
+```python
+--8<-- "docs/_code/quickstart.py::9"
+```
+
+This code snippet prepares a file `demo.txt` on your machine, ready to be added to the lakeFS repository, so let's do just that:
+
+```python
+--8<-- "docs/_code/quickstart.py:12:17"
+```
+
+If you execute the `quickstart.py` script at this point, you can already see the [committed file](http://localhost:8000/repositories/repo/object?ref=main&path=demo.txt) in the lakeFS web UI:
+
+![](_images/quickstart-lakefs-ui.png)
+
+While examining the file contents in the browser is nice, we want to access the committed file programmatically. Add the following lines at the end of your script and observe the output:
+
+```python
+--8<-- "docs/_code/quickstart.py:20:21"
+```
+
+Note that executing the same code multiple times will only result in a single commit in the repository since the contents of the file on disk and in the repository are identical.
+
+In addition to simple read and write operations, the `fsspec` file system interface also allows us to list the files in a repository folder using `ls`, and query the metadata of objects in the repository through `info` (akin to the POSIX `stat` system call).
+Let's add the following code to our script and observe the output:
+
+```python
+--8<-- "docs/_code/quickstart.py:23:30"
+```
+
+As the last order of business, let's clean up the repository to its original state by removing the file using the `rm` operation and creating another commit (also, the local file is deleted, since we don't need it anymore):
+
+```python
+--8<-- "docs/_code/quickstart.py:33:36"
+```
+
+!!! success
+
+    You now have all the basic tools available to version data from your Python code using the file system interface provided by `lakefs-spec`.
+
+??? tip "Full example code"
```

Comment:
I propose not to provide full code but rather piece by piece only - as a user needs to see outcomes after each step (the last steps removes committed file - so if run in "one go" no change actually to be seen).

Reply:
See above. I would still keep the full example around, since it allows the user to look at the full code in a single location. (doesn't really matter that the file is deleted again at the end of the script, that's expected and explained behavior of the script after all)